src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/convert_pagedattn_inputs.cpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "convert_pagedattn_inputs.hpp"
+
+#include <cstdint>
+#include <memory>
+#include <transformations/utils/gen_pattern.hpp>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/paged_attention.hpp"
+#include "openvino/util/log.hpp"
+#include "transformations/utils/utils.hpp"
+using namespace ov::gen_pattern;
+
+ov::intel_cpu::ConvertPagedAttnInputs::ConvertPagedAttnInputs(const Config& config) : m_config(config) {
```

Comment:
Given we have the same task for GPU ([CVS-161325](https://jira.devtools.intel.com/browse/CVS-161325)), it worth to make the pass common and reuse inside GPU Plugin.  Ideally it should be enabled by default in Common transformation, as external pipeline (like genai) expects specified precision for any device.
@sshlyapn Could you please take a look on this pass? I bet GPU has some specifics for default values, so lets decide how ConvertPagedAttnInputs transformation should be properly parametrized.

Reply:
@sshlyapn If make this as a common transformation, I think we could have a parameter passed to the transformation like
```cpp
KVCahceConfig {
  ov::element::Type key_cache_precision,
  ov::element::Type value_cache_precision,
  size_t key_cache_block_size,
  size_t value_cache_block_size,
  size_t key_cache_group_size,
  size_t value_cache_group_size,
  bool key_cache_quant_bychannel,
  bool value_cache_quant_bychannel,
}
```

Reply:
@zhangYiIntel this config looks fine to me. Just one note: in GPU, we use different dims order for key and value (head_size and block_size are swapped for key), so it would be great to add an additional "key_/value_ cache_dims_order" parameter to be able to set arbitrary dims ordering
Also, maybe we can make some configurable `void adjust_cache_values(const ov::element::Type precision, const size_t group_size, const bool bychannel, int64_t& head_size, int64_t& block_size)` to let plugins apply their own rules for head/block sizes adjustment